pennylane/devices/default_qubit_jax.py

```diff
@@ -231,75 +240,64 @@ def states_to_binary(samples, num_wires, dtype=jnp.int32):
             List[int]: basis states in binary representation
         """
         powers_of_two = 1 << jnp.arange(num_wires, dtype=dtype)
-        states_sampled_base_ten = samples[:, None] & powers_of_two
```

Comment:
Sorry for the massive changes in this area. They are analogue to the ones in `DefaultQubit.estimate_probability` and also prepare shot vector support.